lib/Remotery.c

```diff
@@ -8820,4 +8817,92 @@ RMT_API void _rmt_EndMetalSample(void)
 
 #endif // RMT_USE_METAL
 
+// Iterator
+RMT_API rmtSampleIterator _rmt_IterateChildren(rmtSample* sample)
+{
+    rmtSampleIterator iterator;
+    iterator.sample = 0;
+    iterator.initial = sample != NULL ? sample->first_child : 0;
+    return iterator;
+}
+
+RMT_API rmtBool _rmt_IterateNext(rmtSampleIterator* iter)
+{
+    if (iter->initial != NULL)
+    {
+        iter->sample = iter->initial;
+        iter->initial = 0;
+    }
+    else
+    {
+        if (iter->sample != NULL)
+            iter->sample = iter->sample->next_sibling;
+    }
+
+    return iter->sample != NULL ? RMT_TRUE : RMT_FALSE;
+}
```

Comment:
Following the pattern from this gist:
https://gist.github.com/pervognsen/744066869da509a15c4e08fc9408207e